src/test/java/com/taboola/rest/api/internal/serialization/SerializationMapperCreatorTest.java

```diff
@@ -68,17 +70,82 @@ public void createObjectMapper_serializationConfigWithIgnoreAnySetterAndApiObjec
         Assert.assertNull("Name is parsed incorrectly", sampleApi.name);
     }
 
+    @Test
+    public void createObjectMapper_defaultSerializationConfigAndApiObjectHasValidEnum_enumParsedCorrectly() throws IOException {
+        SerializationConfig serializationConfig = new SerializationConfig();
+        ObjectMapper objectMapper = SerializationMapperCreator.createObjectMapper(serializationConfig);
+
+        SampleApi sampleApi = objectMapper.readValue("{ \"letter\": \"B\" }", SampleApi.class);
+
+        Assert.assertEquals("Unknown enum is parsed incorrectly", SampleEnum.B, sampleApi.letter);
+    }
+
+    @Test
+    public void createObjectMapper_defaultSerializationConfigAndApiObjectHasInvalidEnum_defaultValueUsedOnSerialization() throws IOException {
+        SerializationConfig serializationConfig = new SerializationConfig();
+        ObjectMapper objectMapper = SerializationMapperCreator.createObjectMapper(serializationConfig);
+
+        SampleApi sampleApi = objectMapper.readValue("{ \"letter\": \"D\" }", SampleApi.class);
+
+        Assert.assertEquals("Unknown enum is parsed incorrectly", SampleEnum.UNKNOWN, sampleApi.letter);
+    }
+
+    @Test
+    public void createObjectMapper_defaultSerializationConfigAndApiObjectHasEmptyEnumValue_defaultValueUsedOnSerialization() throws IOException {
+        SerializationConfig serializationConfig = new SerializationConfig();
+        ObjectMapper objectMapper = SerializationMapperCreator.createObjectMapper(serializationConfig);
+
+        SampleApi sampleApi = objectMapper.readValue("{ \"letter\": \"D\" }", SampleApi.class);
+
+        Assert.assertEquals("Unknown enum is parsed incorrectly", SampleEnum.UNKNOWN, sampleApi.letter);
+    }
+
+    @Test
+    public void createObjectMapper_serializationConfigWithReadUnknownEnumsDisabledAndApiObjectHasValidEnum_enumParsedCorrectly() throws IOException {
+        SerializationConfig serializationConfig = new SerializationConfig().setShouldDisableReadUnknownEnumValuesAsDefaultValue();
+        ObjectMapper objectMapper = SerializationMapperCreator.createObjectMapper(serializationConfig);
+
+        SampleApi sampleApi = objectMapper.readValue("{ \"letter\": \"B\" }", SampleApi.class);
+
+        Assert.assertEquals("Unknown enum is parsed incorrectly", SampleEnum.B, sampleApi.letter);
+    }
+
+    @Test(expected = InvalidFormatException.class)
```

Comment:
found [this link](https://www.baeldung.com/junit-assert-exception) on how to write tests that have exceptions.

Unsure if this is the actual practice though. I assume it could also be implemented with a try catch.

Reply:
In current junit version you can not really check exception message if written like this so I personally prefer try catch.
So it depends if you want to check exception message or not.